internal/engine/engine_test.go

```diff
@@ -138,6 +148,75 @@ var _ = Describe("Execute Checks tests", func() {
 				Expect(err).To(HaveOccurred())
 			})
 		})
+		Context("it is a bundle made with GNU tar layer", func() {
+			BeforeEach(func() {
+				var buf bytes.Buffer
+
+				err := writePaddedTarball(&buf, []byte("mycontent"), "myfile", 10)
+				Expect(err).ToNot(HaveOccurred())
+
+				layer := static.NewLayer(buf.Bytes(), types.MediaType("application/vnd.docker.image.rootfs.diff.tar"))
+				img, err := mutate.AppendLayers(empty.Image, layer)
+				Expect(err).ToNot(HaveOccurred())
+
+				src = fmt.Sprintf("%s/test/crane", u.Host)
+
+				err = crane.Push(img, src)
+				Expect(err).ToNot(HaveOccurred())
+			})
+			It("should just hang forever err and hash mean nothing", func() {
```

Comment:
Does this really hang forever?

Reply:
No it does not, I wrote this test against `main`, which would be true for `main` but not with the fix. I'll double check the asserts again so they are correct.